### PR TITLE
feat: Add panic hook for resetting the terminal

### LINKED ISSUE #104: Add a panic hook to reset terminal upon panic
**Is your feature request related to a problem? Please describe.**

When an unexpected error happens, the terminal is messed up.

**Describe the solution you'd like**

Add a custom panic hook. See: https://github.com/ratatui-org/rust-tui-template/pull/13

> The code changes on this PR will add a panic hook, so if a problem happens and the program panics, before the program closes it will run the code inside the panic hook that will leave alternate screen and disable raw mode. This way, you won't have your terminal messed up if an unexpected error happens.

**Describe alternatives you've considered**

None.

**Additional context**

The implementation might be the same as the linked pull request above.

### PR DESCRIPTION
**Pull request by PiotrekB416**
## Description

Added a panic hook following the [raratui implementation](https://github.com/ratatui-org/rust-tui-template/pull/13).
TL;DR When an unexpected error happens program exits alternate screen and disables raw mode instead of messing up the terminal.

## Motivation and Context

Fixes #104 

## How Has This Been Tested?

I used cargo test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other 

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [x] I have formatted the code with [rustfmt](https://github.com/rust-lang/rustfmt).
- [x] I checked the lints with [clippy](https://github.com/rust-lang/rust-clippy).
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed. (Some test fail, but they failed before I made any changes)

### CODE REVIEW

File: src/term/tui.rs [hunk 1]
```diff
@@ -95,13 +104,19 @@ impl<B: Backend> Tui<B> {
...
+		Self::reset()?;
+		self.terminal.show_cursor()?;
+		Ok(())
+	}
+
+	fn reset() -> Result<()> {
 		terminal::disable_raw_mode()?;
 		crossterm::execute!(
-			io::stderr(),
+			io::stdout(),
```
Comment (orhun) [inline src/term/tui.rs:115]: I think this should be `stderr` instead since the terminal runs on `stderr`:

https://github.com/orhun/gpg-tui/blob/7975a47ab995cf5fddbcd49cf23ea399e46e085f/src/main.rs#L37

**Comment by orhun** [comment]
Thank you! 🐻
